src/main/java/pp/mvc/products/ProductController.java

```diff
@@ -0,0 +1,77 @@
+package pp.mvc.products;
+
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Controller
+public class ProductController {
+
+    public ProductRepository productRepository;
+
+    public ProductController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping("/")
+    public String home(Model model) {
+        List<String> list = new ArrayList<>();
+        list.add("WSZYSTKIE");
+        list.add("SPOŻYWCZE");
+        list.add("DOMOWE");
+        list.add("INNE");
+
+        model.addAttribute("home", list);
+        model.addAttribute("product", new Product());
+        model.addAttribute("products", productRepository.getAll());
+        return "home";
+    }
+
+
+    @GetMapping("/lista")
+    public String all(Model model, @RequestParam(required = false, name = "kategoria") Category category) {
```

Comment:
Zauważ, że `WSZYSTKIE` Ci nie działa, ponieważ (słusznie) zwracasz wszystkie produkty tylko w momencie jak nie ma podanej żadnej kategorii. Nie potrzebujesz zatem posiadać enuma `WSZYSTKIE` bo dla Ciebie wszystkie produkty to kategoria = null. Zresztą zobacz, że sama oznaczyłaś ją jako opcjonalna. Tak więc wystarczy, że w HTML'u na linka który się nazywa `WSZYSTKIE`. Przekierujesz użytkownika na URL'a `/lista` bez podawania parametru `kategoria`.